python/datawrangling/cms-openpayment/general_payment_process_manufacturer.py

```diff
@@ -0,0 +1,112 @@
+"""
+This script attempts to create a canonical name for the names of the manufacturers 
+and GPOs. The script first checks whether the mapping between ID and name/state/country
+information is consistent, then converts the names into a canonical form and outputs
+a csv containing this information.
+
+Inconsistency in raw data can be handled by specifying a yaml file as the dict 
+parameter. This yaml file should contain a sequence of dictionaries, each dictionary
+containing at least the id key. Information in this file is written directly to the 
+output csv. 
+
+author: sangxia
+"""
+
+import pandas as pd
+import unicodedata
+import numpy as np
+import csv
+import argparse
+import yaml
+import re
+import copy
+
+def name_canon(s):
+    ret = unicodedata.normalize('NFKD',s.strip()).encode('ascii','ignore').decode()
+    ret = re.sub('  +', ' ', ret)
+    return str.upper(ret.strip())
+
+def process_manufacturers(infile, outfile,name_dict):
+    df = pd.read_csv(infile)
+    colid = 'Applicable_Manufacturer_or_Applicable_GPO_Making_Payment_ID' 
+    colname = 'Applicable_Manufacturer_or_Applicable_GPO_Making_Payment_Name'
+    colstate = 'Applicable_Manufacturer_or_Applicable_GPO_Making_Payment_State'
+    colcountry = 'Applicable_Manufacturer_or_Applicable_GPO_Making_Payment_Country'
+    df = df.loc[~df[colid].isin(name_dict)].drop_duplicates()
+#    print('Checking null values')
+#    for c in df.columns:
+#        if c is not colstate:
+#            assert(df[c].isnull().sum()==0)
+#    assert(((df[colstate].isnull()) & (df[colcountry] == 'United States')).sum()==0)
+#    assert((df.loc[df[colstate].notnull(), colstate].str.len() != 2).sum()==0)
+    print('Computing canonical names')
+    df['Name_canonical'] = df[colname] \
```

Comment:
Just a personal preference on my part, but I'd break lines here such that each "step" is on its own line, like so:

```
df['Name_canonical'] = df[colname] \
    .str.normalize('NFKD') \
    .str.encode('ascii','ignore') \
    .str.decode('utf-8') \
    .str.upper() \
    .str.replace(' +', '') \
    .str.replace('[^A-Z0-9]','') \
    .str.strip()
```